src/Analyzers/CSharp/Analyzers/SimplifyLinqExpressions/CSharpSimplifyLinqExpressionsDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Operations;
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+#if CODE_STYLE
+using OptionSet = Microsoft.CodeAnalysis.Diagnostics.AnalyzerConfigOptions;
+#endif
+
+namespace Microsoft.CodeAnalysis.CSharp.SimplifyLinqExpressions
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpSimplifyLinqExpressionsDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+
+        public CSharpSimplifyLinqExpressionsDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.SimplifyLinqExpressionsDiagnosticId,
+                  option: null,
+                  title: new LocalizableResourceString(nameof(CSharpAnalyzersResources.Simplify_linq_expressions), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        private void OnCompilationStart(CompilationStartAnalysisContext context)
+        {
+            var whereMethods = ImmutableHashSet.CreateBuilder<IMethodSymbol>();
+            var linqMethods = ImmutableHashSet.CreateBuilder<IMethodSymbol>();
+            var validLinqCalls = new List<string> { "First", "Last", "Single", "Any", "Count", "SingleOrDefault", "FirstOrDefault", "LastOrDefault" };
+
+            var enumNamedType = context.Compilation.GetTypeByMetadataName("System.Linq.Enumerable");
+            var queryNamedType = context.Compilation.GetTypeByMetadataName("System.Linq.Queryable");
```

Comment:
I thought in the design meeting it was decided IQueryable would not be changed? I might be remembering wrong.

Reply:
that's my recollection.  we already know there are Queryable impls that don't have tehse semantics.  so we should not do this.

Reply:
Sorry I misunderstood the last few comments from the design meeting at first because I was confused about the expression trees (just went back and re-watched), I'll update accordingly